src/NzbDrone.Core/Download/Clients/Flood/Flood.cs

```diff
@@ -0,0 +1,214 @@
+                            result.Add(remoteEpisode.ParsedEpisodeInfo.Quality.Quality.ToString());
+                            break;
+                        case (int)AdditionalTags.Language:
+                            result.Add(remoteEpisode.ParsedEpisodeInfo.Language.ToString());
+                            break;
+                        case (int)AdditionalTags.ReleaseGroup:
+                            result.Add(remoteEpisode.ParsedEpisodeInfo.ReleaseGroup);
+                            break;
+                        case (int)AdditionalTags.Year:
+                            result.Add(remoteEpisode.Series.Year.ToString());
+                            break;
+                        case (int)AdditionalTags.Indexer:
+                            result.Add(remoteEpisode.Release.Indexer);
+                            break;
+                        case (int)AdditionalTags.Network:
+                            result.Add(remoteEpisode.Series.Network);
+                            break;
+                        default:
+                            throw new DownloadClientException("Unexpected additional tag ID");
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public override string Name => "Flood";
+
+        protected override string AddFromTorrentFile(RemoteEpisode remoteEpisode, string hash, string filename, byte[] fileContent)
+        {
+            _proxy.AddTorrentByFile(Convert.ToBase64String(fileContent), HandleTags(remoteEpisode, Settings), Settings);
+
+            return hash;
+        }
+
+        protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string hash, string magnetLink)
+        {
+            _proxy.AddTorrentByUrl(magnetLink, HandleTags(remoteEpisode, Settings), Settings);
+
+            return hash;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            var items = new List<DownloadClientItem>();
+
+            var list = _proxy.GetTorrents(Settings);
+
+            foreach (var torrent in list)
+            {
+                var properties = torrent.Value;
+
+                if (!Settings.Tags.All(tag => properties.Tags.Contains(tag)))
+                {
+                    continue;
+                }
+
+                var item = new DownloadClientItem
+                {
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this),
+                    DownloadId = torrent.Key,
+                    Title = properties.Name,
+                    OutputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(properties.Directory)),
+                    Category = properties.Tags.Count > 0 ? properties.Tags[0] : null,
+                    RemainingSize = properties.SizeBytes - properties.BytesDone,
+                    RemainingTime = TimeSpan.FromSeconds(properties.Eta),
```

Comment:
This is returning `-1` for paused torrents, instead we should return null (like we do for other clients): `properties.Eta > 0 ? (TimeSpan?)(TimeSpan.FromSeconds(properties.Eta)) : null,` will do it.